tests/test_frame.py

```diff
@@ -2363,3 +2364,16 @@ def test_frame_dropna_subset() -> None:
         assert_type(df.dropna(subset=df.columns.drop("col1")), pd.DataFrame),
         pd.DataFrame,
     )
+
+
+def test_npint_loc_indexer() -> None:
+    # GH 508
+
+    df = pd.DataFrame(dict(x=[1, 2, 3]), index=np.array([10, 20, 30], dtype="uint64"))
+
+    def get_NDArray(df: pd.DataFrame, key: npt.NDArray[np.uint64]) -> pd.DataFrame:
+        df2 = df.loc[key]
```

Comment:
Wouldn't any `np.NDArray` work (not just integer) as long as the DataFrame index is of the same type?

Reply:
could probably only enforce a tight dtype match if index was generic

Reply:
> Wouldn't any `np.NDArray` work (not just integer) as long as the DataFrame index is of the same type?

Probably, but since we can't track the `dtype` of an `Index` in a `DataFrame`, I'm limiting this for now to the issue that was reported.  I think most people use arrays of `int` or arrays of `str` (which we probably could add), but I'd rather be incremental in adding support for all the possible types.